fourcc.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+fourcc_include = sys.argv[1]
+
+fmt_list = []
+mod_list = []
+
+fmt = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?fourcc_code', flags=re.M)
+mod = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?fourcc_mod_code', flags=re.M)
```

Comment:
This missing some modifiers like `DRM_FORMAT_MOD_BROADCOM_SAND256`.

```c
#define DRM_FORMAT_MOD_BROADCOM_SAND256 \
    DRM_FORMAT_MOD_BROADCOM_SAND256_COL_HEIGHT(0)
```

Reply:
Those should probably be added back as a special case.
```python
mod_broadcom = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?DRM_FORMAT_MOD_BROADCOM', flags=re.M)

Reply:
Why not match using the `DRM_FORMAT_MOD_` prefix (and add a special case for i915 which has a different prefix)?

Reply:
I'm trying to be somewhat defensive about what modifiers will be added in the future. We already have things like `AFBC_FORMAT_MOD_*` which are not modifiers themselves, but flags to another macro, so matching `_FORMAT_MOD_` isn't reliable. I'm just waiting for
```c
#define DRM_FORMAT_MOD_VENDOR_FLAG (1ULL)
```
to happen.

I think matching against `fourcc_mod_code` is a more reliable way to truly match things that are modifiers, but that doesn't match against macros that are defined in terms of other macros.

Reply:
> `mod_broadcom = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?DRM_FORMAT_MOD_BROADCOM', flags=re.M)`

This doesn't seem to match anything.

Reply:
> This missing some modifiers like `DRM_FORMAT_MOD_BROADCOM_SAND256`.

I was able to get results using
`mod_broadcom = re.compile(r'^#define (DRM_FORMAT_MOD_BROADCOM\w+[^\w(])', flags=re.M)`

@emersion Can you see if the latest iteration does The Right Thing?

Reply:
The latest iteration's output LGTM. It does add some formats we were missing.